### PR TITLE
Bug 1643651 - Add DoH CFR message

### PR DESCRIPTION
**Pull request by piatra**

### CODE REVIEW

File: schema/messaging-system-special-message-actions.schema.json [hunk 1]
```diff
@@ -13,9 +13,7 @@
     "description": "Disables all STP doorhangers.",
     "properties": {
       "type": {
-        "enum": [
-          "DISABLE_STP_DOORHANGERS"
-        ],
+        "enum": ["DISABLE_STP_DOORHANGERS"],
```
Comment (piatra) [inline schema/messaging-system-special-message-actions.schema.json:16]: Prettier over did it.

File: cfr.json [hunk 2]
```diff
@@ -1707,5 +1707,60 @@
...
+      "layout": "icon_and_message",
+      "text": {
+        "string_id": "cfr-doorhanger-doh-body"
+      },
+      "icon": "chrome://browser/skin/connection-secure.svg",
+      "buttons": {
+        "secondary": [
+          {
+            "label": {
+              "string_id": "cfr-doorhanger-doh-secondary-button"
```
Comment (ncloudioj) [inline cfr.json:1727]: Just to confirm, this seems like a new string (not yet added in the remote FTL), we will have to wait for the next remote FTL publish to ship it, correct?

File: cfr.yaml [hunk 3]
```diff
@@ -1291,3 +1291,39 @@
     lifetime: 3
   targeting: firefoxVersion >= 74 && localeLanguageCode == 'en' &&
     userPrefs.cfrFeatures && region == 'US'
+- id: DOH_ROLLOUT_CONFIRMATION
+  targeting: "'doh-rollout.enabled'|preferenceValue && !'doh-rollout.disable-heuristics'|preferenceValue && !'doh-rollout.skipHeuristicsCheck'|preferenceValue && !'doh-rollout.doorhanger-decision'|preferenceValue && firefoxVersion >= 79"
```
Comment (ncloudioj) [inline cfr.yaml:1295]: I can't confirm the DoH related prefs in this targeting to my knowledge, do we have a reference for those prefs?

File: cfr.json (hunk 2 above)
Comment (piatra) [reply cfr.json:1727]: It's already in central https://searchfox.org/mozilla-central/rev/a87a1c3b543475276e6d57a7a80cb02f3e42b6ed/browser/locales/en-US/browser/newtab/asrouter.ftl#272

Comment (ncloudioj) [reply cfr.json:1727]: Yep, I saw those strings. Just wanted to confirm it :)

File: cfr.yaml (hunk 3 above)
Comment (piatra) [reply cfr.yaml:1295]: Code: https://firefox-source-docs.mozilla.org/toolkit/components/messaging-system/docs/TriggerActionSchemas/TriggerActionSchemas.html
Doc discussion: https://docs.google.com/document/d/184dUI1bMauDWkxcm94tKouyiOWNZ7QGhfnOqYTRf3P4/edit#heading=h.5840t7jz7l1f

Comment (ncloudioj) [reply cfr.yaml:1295]: Thanks! That's very helpful.
🚢

**Pull request merged**